autoware_map_msgs/srv/LoadDifferentialPointCloudMap.srv

```diff
@@ -0,0 +1,13 @@
+# Area query for map loading
+# (PCD grids that overlaps with the area is going to be loaded)
+AreaInfo area
+
+# The IDs of PCD maps that the node already has
+string[] already_loaded_ids
+
+---
+# Newly loaded PCD maps with ID
+PointCloudMapWithID[] loaded_pcds
+
+# Map IDs that the client side should remove
+string[] ids_to_remove
```

Comment:
Why are we communicating list of ids to remove?

I think it is better to tell the client what id's to use for the given AreaInfo query instead of the `ids_to_remove`.

Reply:
And client can decide what to remove itself, if it wants to.

Reply:
I and @.yukkysaito had a short discussion on the topic here: https://github.com/autowarefoundation/autoware_msgs/pull/39#discussion_r981918593.

It is true that providing `what id's to use for the given AreaInfo query` would enable the clients adopt more flexible implementation, but we were not sure if this flexibility is currently needed. I believe that it is better to use `ids_to_remove` as long as we don't have any specific use cases in which the client wants `what id's to use` instead of `what id's to remove`.

Do you have any ideas on this? (e.g. any such concrete use case)

Reply:
- It makes more sense because the client asked for id's in an area and it would return these list of id's. (It's already known by server, easier to implement)
- `what id's to use ` would provide the same functionality as `what id's to remove` but more. Client can choose to remove id's that are not returned.
- If client relies on these responses to remove id's, the server needs to remember what else it has provided to each client, it adds unnecessary complexity.

Reply:
@kminoda any opinions on this?

Reply:
Also after thinking about this, even `what id's to use` is unnecessary because the client can just iterate through the `PointCloudMapWithID[] loaded_pcds` so I think we can completely remove this item.

Reply:
@kminoda 

> `loaded_pcds` is merely $X_1 \backslash X_0$, and thus the client would not be able to reconstruct the set $X_0 \backslash X_1$. 

Yeah I was wrong here sorry. But I wanted to say that ( $X_1$ = what IDs to use) and we would provide this back to the client. $X_0 \cap X_1$ = what id's to keep

![Untitled](https://user-images.githubusercontent.com/10751153/195592688-b273260f-52fb-4de9-b34d-a9bdccc28b34.png)

> - Case A (what IDs to remove): The client has $X_0$, $X_1 \backslash X_0$, and $X_0 \backslash X_1$
> - Case B (what IDs to keep): The client has $X_0$, $X_1 \backslash X_0$, and $X_0 \cap X_1$

I will add:
- Case C (queried IDs): The client has $X_0$, $X_1$ with these, complements or intersections can be calculated by the client.

OK so, both A and B would work but you prefer A so that client won't have to do the operation:
- $X_1 \backslash X_0 = X_0 \backslash (X_0 \cap  X_1)$
- (`id's to remove` = `cached ids` - `id's to keep`)

Since they both achieve the same result, we can do it with Case A.

Also with C you can just calculate $X_0 \cap X_1$ too but it's not that important.

Thanks for explaining it.

Reply:
Oh, OK, now I got your point. And thanks for the draw! Then I think we will move on with case A.

FYI, I think Case C would almost will look like Case B in practical. Since we are loading new PCDs only for `new IDs`, the response will look like
- new IDs (all of which attached with a corresponding PCDs)
- IDs to keep
which is the same as Case B.

Reply:
Yes, I will go over the PR one last time and then let's merge it!